### DISCUSSION TITLE
is the diagnostic plots look alright...

### DISCUSSION THREAD

**Discussion by hainiwen** [nlmixr2/nlmixr2#225] [Q&A]
Hi all,

I'm new to nlmixr2 and SAEM, and I'm not sure about how should my traceplots look like.

I'm working on a PBPK model with 16 compartments, whereas datset is a bit sparse. I read from many MONOLIX examples that traceplots always stabilize in the very end, however mine looks like this:

![image](https://github.com/nlmixr2/nlmixr2/assets/56790121/9cb6ea37-027b-4975-bad4-634ced0ad85e)

My questions are:

1. is it normal that the traceplots stabilize in one phase but then drop down again?
2. For the right upper corner one the parameter Kpu_lung_tb doesn't not tend to stabilize, it that okay to look like this because this parameters seems to have a high shrinkage? The model reads the last value as its estimates.
3. currently, i use these settings:  saemControl(nBurn = 200, #200 SAEM burn-in iterations (the default)
                     nEm   = 2000, #300 EM iterations (the default)
                     print = 50,
                     tol = 1e-06), 
         tableControl(cwres = TRUE,npde=TRUE) #calculates NONMEM-style conditional weighted residuals and npde for diagnostics
  )
And i increased nEM from 500 to 2000, i didn't quite feel big improvements, and if i went even upper, R sometimes crash itself. What should I do to improve the estimation? 

4. Any recommended readings for SAEM as an algorithm and how to diagnose it?

Thank you for your help in advance!!
Regards,

Haini

**Comment by mattfidler** [comment]
Hi @hainiwen 

Usually PBPK models only have a few between subject variabilities (if any) and are often fit better with `focei`.  You seem to have many.  

My guess this is highly over-parameterized.  As a start, I would try to remove the between subject variabilities of the parameters that are not stabilizing and see if the fit stabilizes with these items removed.

I'm unsure where a practical guide to the `saem` algorithm would be.  Most of the items I'm aware of are more related to the math and application of the algorithm.

**Reply by hainiwen** [reply]
Thanks for your suggestions!
 I thought that since my observations were quite sparse, SAEM worth a try? Why do you think FOCEI is better ?.... 

Yea my tricky question is exactly to get variabilities out of a complex PBPK model,,, guess i have to opt out one of them ?

**Reply by hainiwen** [reply]
And yes, once BSV is removed, for example, for Kpu_lung_tb here, the parameters stabilize somewhere in between the iterations. 
![1](https://github.com/nlmixr2/nlmixr2/assets/56790121/e23234c7-1f35-4b7c-a04f-9d53e2dcfe1a)


How do you think about this one...

**Reply by mattfidler** [reply]
Still hasn't converged (though it looks better).

**Reply by mattfidler** [reply]
> Thanks for your suggestions! I thought that since my observations were quite sparse, SAEM worth a try? Why do you think FOCEI is better ?....
> 
> Yea my tricky question is exactly to get variabilities out of a complex PBPK model,,, guess i have to opt out one of them ?


- `focei` works better with a small number of between subject variabilities. 
- `nlminb` and `nlm` work better with no between subject variabilities (and will give parameter uncertainty with a successful covariance step).
- `saem` works but may have convergence issues (as observed above). It is also more likely to support over-parametrized models.

**Reply by hainiwen** [reply]
Thank you so much, I'll try nlminb next step.
And I also cut the model down a bit with less parameters and compartments, this is what I got: 
and I wonder, is it okay for error part to be unstable like that or I'm still not getting convergence...


![image](https://github.com/nlmixr2/nlmixr2/assets/56790121/28ec919d-f23f-4fbd-b2e7-44b1ec67233a)
